internal/manifest/parse.go

```diff
@@ -33,14 +40,38 @@ func NewPathExtractor(cache *filemutex.MutexCache) *PathExtractor {
 	return &PathExtractor{fileMutexCache: cache}
 }
 
-func (p PathExtractor) GetPathFromRawManifest(ctx context.Context,
+func (p PathExtractor) FetchLayerToFile(ctx context.Context,
+	imageSpec v1beta2.ImageSpec,
+	keyChain authn.Keychain,
+	layerName string,
+) (string, error) {
+	switch imageSpec.Type {
+	case v1beta2.OciRefType:
+		return p.getPathForFetchedLayer(ctx, imageSpec, keyChain, v1beta2.RawManifestLayerName+".yaml")
+	case v1beta2.OciDirType:
+		tarFile, err := p.getPathForFetchedLayer(ctx, imageSpec, keyChain, layerName+".tar")
```

Comment:
This logic here should be improved, you should not always decompress the content, instead:
1. download the layer as `raw-manifest.tar`, but only once, this is already done in `getPathForFetchedLayer`
2. check if the decompressed file `raw-manifest.yaml` exits, if exists, return
3. if not exists, decompress the file to a folder